[stdlib] Do not copy elements when using `FileHandle.read_bytes()`

MoSafi2 opened the pull request:
Fixes #2051, allows `FileHandle.read_bytes()` to return `List[Int8]` without additional copies, depends on #2182.

MoSafi2 committed:
Fixes [stdlib] Do not copy elements when using FileHandle.read_bytes()

Signed-off-by: Mohamed Mabrouk <mohamed.mabrouk@rwth-aachen.de>

JoeLoser reviewed (approved):
Looks good, thank you! Please see my comment about cleaning up the test now that we're using `assert_equal` 👍

MoSafi2 committed:
[mojo-stdlib] removing unneccassry print statement in

Signed-off-by: Mohamed Mabrouk <mohamed.mabrouk@rwth-aachen.de>

MoSafi2 committed:
fix mojo formatting

Signed-off-by: Mohamed Mabrouk <mohamed.mabrouk@rwth-aachen.de>

gabrieldemarmiesse committed:
[stdlib] Remove FileCheck in `test_paramenv.mojo` (#2302)

Related to https://github.com/modularml/mojo/issues/2024

Signed-off-by: Gabriel de Marmiesse <gabrieldemarmiesse@gmail.com>

gabrieldemarmiesse committed:
[stdlib] Remove FileCheck from `test_windows_target.mojo` (#2300)

Related to https://github.com/modularml/mojo/issues/2024

Signed-off-by: Gabriel de Marmiesse <gabrieldemarmiesse@gmail.com>

lsh committed:
[mojo-stdlib] `Optional.value()` returns a `Reference` (#2226)

Change `Optional.value()` to return a `Reference` to `T` rather than `T`.  

Fixes https://github.com/modularml/mojo/issues/2179.

Signed-off-by: Lukas Hermann <lukashermann28@gmail.com>

gabrieldemarmiesse committed:
[stdlib] Fix assert_equal[SIMD]() not raising if any elements are equal (#2279)

To reproduce the bug:
```mojo
with assert_raises():
    assert_equal(SIMD[DType.uint8, 2](1, 1), SIMD[DType.uint8, 2](1, 2))
```

This is because using `!=` returns `[False, True]` then, when using `.__bool__()` on this, since `__bool__()` is equivalent to a `reduce_and()`, it returns False, and the error is not triggered.

Long story short, doing `if a != b` doesn't do what one would expect. This bug is not present with `assert_not_equal()`.

Signed-off-by: gabrieldemarmiesse <gabrieldemarmiesse@gmail.com>

gabrieldemarmiesse committed:
[stdlib] Remove FileCheck in `test_intrinsics.mojo` (#2285)

Related to #2024 

Signed-off-by: Gabriel de Marmiesse <gabrieldemarmiesse@gmail.com>

gabrieldemarmiesse committed:
[stdlib] Remove FileCheck in `test_build_info_debug.mojo` (#2291)

Related to #2024

Signed-off-by: Gabriel de Marmiesse <gabrieldemarmiesse@gmail.com>

gabrieldemarmiesse committed:
[stdlib] Remove FileCheck in `test_bitcast.mojo` (#2296)

Related to #2024

Signed-off-by: Gabriel de Marmiesse <gabrieldemarmiesse@gmail.com>

gabrieldemarmiesse committed:
[stdlib] Remove FileCheck in `test_has_intel_amx.mojo` (#2288)

Related to #2024

Signed-off-by: Gabriel de Marmiesse <gabrieldemarmiesse@gmail.com>

MoSafi2 committed:
fix imports; reformat

Signed-off-by: Mohamed Mabrouk <mohamed.mabrouk@rwth-aachen.de>
